### PR TITLE
[ABW-2532] Fix issue with 3 letter bip39 word when losing focus

### PR DESCRIPTION
**Pull request by CyonAlexRDX**
Fixes https://github.com/radixdlt/babylon-wallet-ios/issues/820

# Description
This PR fixes a bug where some 3 letter words are not counted as valid when TextField loses focus, e.g. "add", which is a valid BIP39 word. "add" has some _word friends_ like "addict", also starting with "add", the bug was that since "add" was not unambigously identified (due to friend "addict") we incorrectly did not set the `ImportMnemonicWord.value` to `.complete` upon losing focus.

Apart from the fix, some unit tests capturing this has been added.

https://github.com/radixdlt/babylon-wallet-ios/assets/116169792/92bf3d1e-d55b-41e5-aac5-f3fded67837f

### CODE REVIEW

File: RadixWallet/Features/SettingsFeature/ImportFromOlympiaLegacyWallet/Coordinator/ImportOlympiaWalletCoordinator+View.swift [hunk 1]
```diff
@@ -51,9 +51,7 @@ extension ImportOlympiaWalletCoordinator.Path {
 						/State.importMnemonic,
 						action: Action.importMnemonic,
 						then: { childStore in
-							NavigationView {
-								ImportMnemonic.View(store: childStore)
-							}
+							ImportMnemonic.View(store: childStore)
```
Comment (CyonAlexRDX) [inline RadixWallet/Features/SettingsFeature/ImportFromOlympiaLegacyWallet/Coordinator/ImportOlympiaWalletCoordinator+View.swift:54]: @kugel3 reverting the change you made recently, which broke toolbar in Import from Olympia, which combined with the 3 letter bug, prevents users from inputting some mnemonics (containing 3 letter words which are ambigious with 3 letters)

Do you remember why you made the change? relating to navigation title perhaps?

Comment (kugel3) [reply RadixWallet/Features/SettingsFeature/ImportFromOlympiaLegacyWallet/Coordinator/ImportOlympiaWalletCoordinator+View.swift:54]: Yes, there should be a navigation view at the top level, it looks very strange without it. But can't we have it _only_ there, and ditch the dummy one?

Comment (Sajjon) [reply RadixWallet/Features/SettingsFeature/ImportFromOlympiaLegacyWallet/Coordinator/ImportOlympiaWalletCoordinator+View.swift:54]: ImportMnemonic is used from many places in the app:
* Olympia migration
* Restore wallet - part of ImportManyMnemonicsFlow
* Import from Account (should be merge with bottom)
* Import from Display Seed Phrases

So having it in ImportMnemonic itself ensure toolbar is always visible disregarding of starting point. But I guess we have no option but to manually ensure it is visible on all places...? Bah.... ought to be an easier way of having a toolbar :/ 

Anyway I will take a look

Comment (CyonAlexRDX) [reply RadixWallet/Features/SettingsFeature/ImportFromOlympiaLegacyWallet/Coordinator/ImportOlympiaWalletCoordinator+View.swift]: fixing toolbar was tricker than I had hoped, so will be done in a separate PR,

File: RadixWallet/Features/ImportMnemonic/ImportMnemonic.swift [hunk 2]
```diff
@@ -310,7 +310,15 @@ public struct ImportMnemonic: Sendable, FeatureReducer {
 
 		case let .word(id, child: .delegate(.lostFocus(displayText))):
 			switch lookup(input: displayText, state) {
-			case .known(.ambigous), .unknown(.notInList):
+			case let .known(.ambigous(candidates, input)):
+				if let exactMatch = candidates.first(where: { $0.word == input }) {
+					state.words[id: id]?.value = .complete(text: displayText, word: exactMatch, completion: ImportMnemonicWord.State.WordValue.Completion.auto(match: .exact))
```
Comment (GhenadieVP) [inline RadixWallet/Features/ImportMnemonic/ImportMnemonic.swift:315]: multiple lines please

**Review by GhenadieVP — APPROVED** [review]

**Pull request merged**